### PR TITLE
Mesos backpressure

### PR DESCRIPTION
**Pull request by ssalinas**
Diff will look nicer once the other mesos_1 PR is merged. Relevant commit is https://github.com/HubSpot/Singularity/commit/d15af0f7b0148388ae6bc5ba5a1cb61e59227eab

Moves the backpressure to the mesos client itself now that 0.2.0 of mesos-rxjava allows us to set backpressure behaviors. Also doubles the default event buffer size to handle large influxes of status updates from reconciliation

/cc @darcatron @baconmania

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java [hunk 1]
```diff
@@ -146,7 +147,14 @@ private void connect(URI mesosMasterURI, FrameworkInfo frameworkInfo, Singularit
...
         .mesosUri(mesosMasterURI)
-        .applicationUserAgentEntry(UserAgentEntries.userAgentEntryForMavenArtifact("com.hubspot.singularity", "SingularityService"));
+        .applicationUserAgentEntry(UserAgentEntries.userAgentEntryForMavenArtifact("com.hubspot.singularity", "SingularityService"))
+        .onBackpressureBuffer(
+            scheduler.getEventBufferSize(),
+            () -> {
+              String message = String.format("Overflow of event buffer (%s), singularity could not keep up!", scheduler.getEventBufferSize());
+              scheduler.onUncaughtException(new EventBufferOverflowException(message));
+            },
+            BackpressureOverflow.ON_OVERFLOW_ERROR);
```
Comment (baconmania) [inline SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java]: So just to confirm, if the set of pending events overflows the max buffer size, we get a top-level exception and abort?

Comment (ssalinas) [reply SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java]: yes, it'll cause an abort if the buffer overflows. I felt that was more in line with our idea of keep state consistent than dropping things. There is an alternative for using an unbounded buffer, but I figured it'd be better to abort with an exception than to oom

Comment (baconmania) [reply SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosSchedulerClient.java]: Agreed.

**Comment by baconmania** [comment]
🚢

**Pull request merged**